b2luigi/batch/processes/htcondor.py

```diff
@@ -0,0 +1,225 @@
+        # take the result path from it
+        shutil.copyfile("settings.json", os.path.join(submit_file_dir, "settings.json"))
+        os.chdir(submit_file_dir)
+        output = subprocess.check_output(cmd, env=curr_env)
+        os.chdir(cur_dir)
+        output = output.decode()
+        match = re.search(r"[0-9]+\.", output)
+        if not match:
+            raise RuntimeError("Batch submission failed with output " + output)
+
+        self._batch_job_id = int(match.group(0)[:-1])
+
+    def kill_job(self):
+        if not self._batch_job_id:
+            return
+        subprocess.check_call(["condor_rm", str(self._batch_job_id)], stdout=subprocess.DEVNULL)
+
+    def _create_htcondor_submit_file(self):
+        output_path = get_output_dirs(self.task, create_folder=True)
+        submit_file_path = os.path.join(output_path, "job.submit")
+        executable_wrapper_path = self._create_executable_wrapper()
+
+        log_file_dir = get_log_file_dir(self.task)
+        stderr_log_file = log_file_dir + "job.err"
+        stdout_log_file = log_file_dir + "job.out"
+        job_log_file = log_file_dir + "job.log"
+
+        general_settings = get_setting("htcondor_settings", dict())
+
+        try:
+            job_settings = self.task.htcondor_settings
+        except AttributeError:
+            job_settings = {}
+
+        with open(submit_file_path, "w") as submit_file:
+            submit_file.write(f"executable = {executable_wrapper_path}\n")
+            submit_file.write("should_transfer_files = YES\n")
+            submit_file.write("transfer_input_files = settings.json\n")
+            submit_file.write(f"log = {job_log_file}\n")
+            submit_file.write(f"output = {stdout_log_file}\n")
+            submit_file.write(f"error = {stderr_log_file}\n")
+
+            for key, item in general_settings.items():
+                submit_file.write(f"{key} = {item}\n")
+
+            for key, item in job_settings.items():
+                submit_file.write(f"{key} = {item}\n")
+
+            submit_file.write(f"queue 1\n")
+
+        return submit_file_path
+
+    def _create_executable_wrapper(self):
+
+        env_setup_path = get_setting("env_setup")
+
+        if not os.path.isfile(env_setup_path):
+            raise FileNotFoundError(f"Environment setup script {env_setup_path} does not exist.")
+
+        output_path = get_output_dirs(self.task, create_folder=True)
+        shell = get_setting("shell", "bash")
+
+        executable_wrapper_path = os.path.join(output_path, "executable_wrapper.sh")
+
+        condor_executable_cmd, _ = self._create_condor_executable_cmd()
+        condor_executable_cmd = " ".join(condor_executable_cmd)
+        with open(executable_wrapper_path, "w") as exec_wrapper:
+            exec_wrapper.write(f"#!/bin/{shell}\n")
+            exec_wrapper.write(f"source {env_setup_path}\n")
+            exec_wrapper.write("echo 'Starting executable'\n")
+            exec_wrapper.write("cd /jwd\n")
```

Comment:
Sorry for bringing this up very late. What is `/jwd`?

Reply:
I guess it's good that you noticed that. It was supposed to be the job's working directory. Seems that this is special to the Bonn cluster. On the NAF this doesn't exist since it uses the submission directory as initial directory when the job starts running.

Reply:
I think you can use a
```
initialdir = {os.getcwd()}
```
in the submission file to work against this. I am currently testing on the NAF and our local system. 
I would go with these changes:
```
diff --git a/b2luigi/batch/processes/htcondor.py b/b2luigi/batch/processes/htcondor.py
index 0126e23..24d4ba1 100755
--- a/b2luigi/batch/processes/htcondor.py
+++ b/b2luigi/batch/processes/htcondor.py
@@ -140,13 +140,6 @@ class HTCondorProcess(BatchProcess):
         submit_file_dir, submit_file = os.path.split(submit_file_path)
         cmd = ["condor_submit", submit_file]
 
-        # apparently you have to be in the same directory as the submit file
-        # to be able to submit jobs with htcondor
-        cur_dir = os.getcwd()
-
-        # have to copy settings file to job working directory since b2luigi has to
-        # take the result path from it
-        shutil.copyfile("settings.json", os.path.join(submit_file_dir, "settings.json"))
         output = subprocess.check_output(cmd, env=self.task_env, cwd=submit_file_dir)
         output = output.decode()
         match = re.search(r"[0-9]+\.", output)
@@ -180,11 +173,11 @@ class HTCondorProcess(BatchProcess):
         with open(submit_file_path, "w") as submit_file:
             submit_file.write(f"""
             executable = {executable_wrapper_path}
-            should_transfer_files = YES
-            transfer_input_files = settings.json
+            initialdir = {os.getcwd()}
             log = {job_log_file}
             output = {stdout_log_file}
             error = {stderr_log_file}
+            transfer_output_files = ""
             """)
 
             for key, item in general_settings.items():
@@ -199,7 +192,7 @@ class HTCondorProcess(BatchProcess):
 
     def _create_executable_wrapper(self):
 
-        env_setup_path = get_setting("env_setup")
+        env_setup_path = os.path.abspath(get_setting("env_setup"))
 
         if not os.path.isfile(env_setup_path):
             raise FileNotFoundError(f"Environment setup script {env_setup_path} does not exist.")
@@ -221,6 +214,6 @@ class HTCondorProcess(BatchProcess):
         st = os.stat(executable_wrapper_path)
         os.chmod(executable_wrapper_path, st.st_mode | stat.S_IEXEC)
 
-        return executable_wrapper_path
+        return os.path.abspath(executable_wrapper_path)
```
That should fix all problems with files and folders...

Reply:
If you use the current folder as start folder, you also do not need to copy the settings file :-)

Reply:
Narf. of course the initaldir does not work...
Anyways, two possibilities: we could just "cd" into the current folder or we can create out own settings file. Will think about it